autosklearn/estimators.py

```diff
@@ -795,14 +797,18 @@ def fit(self, X, y,
         """
         # Before running anything else, first check that the
         # type of data is compatible with auto-sklearn. Legal target
-        # types are: continuous, binary, multiclass.
+        # types are: continuous, continuous-multioutput, 'multiclass'
+        # binary
         target_type = type_of_target(y)
-        if target_type in ['multiclass-multioutput',
-                           'multilabel-indicator',
-                           'unknown',
-                           ]:
-            raise ValueError("regression with data of type %s is not"
-                             " supported" % target_type)
+        supported_types = ['continuous', 'binary', 'multiclass', 'continuous-multioutput']
```

Comment:
Hey, what's the binary and multiclass doing here?

Reply:
I originally thought it was a mistake!, but:
```
type_of_target([1.0, 0.0, 3.0])
'multiclass'
```

And then for binary, the testing expect this:
And also the testing looks for this, for example:
https://github.com/automl/auto-sklearn/blob/859fb993607bd5cfb881aca3b346cf4d1addf470/test/test_automl/test_estimators.py#L233

I am honoring the testing requirement for binary, but I don't agree with it. Should I remove it?
I do believe we need multiclass, because how above scenario treats the floats.

Reply:
As per discussed offline, we will leave binary case (due to 1.0 and 0.0) and multiclass.

I updated the comment  to reflect this